src/components/Container.tsx

```diff
@@ -136,34 +136,48 @@ export class Container<P extends ContainerProperties, S extends ContainerState>
     get placeholderProps(): PlaceHolderModel {
         return {
             cqPath: this.props.cqPath,
-            placeholderClassNames: Constants.NEW_SECTION_CLASS_NAMES,
+            placeholderClassNames: Constants.NEW_SECTION_CLASS_NAMES
         };
     }
 
-    get placeholderComponent() {
+    get placeholderComponent(): JSX.Element | null {
         if (!this.props.isInEditor) {
             return null;
         }
 
         return <ContainerPlaceholder { ...this.placeholderProps }/>;
     }
 
-    public render() {
+    // eslint-disable-next-line max-lines-per-function
+    public render(): JSX.Element {
         let renderScript;
 
-        if (!this.props.isInEditor && this.props.aemNoDecoration){
+        if (!this.props.isInEditor || this.props.aemNoDecoration) {
             renderScript = (
-                <React.Fragment>
-                    { this.childComponents }
-                </React.Fragment>
-            )
+              <React.Fragment>
+                { this.childComponents }
+              </React.Fragment>
+            );
         } else {
+            const ref = React.createRef<HTMLDivElement>();
+
+            if (this.props.isInEditor) {
+                ModelManager.getData({ path: this.props.cqPath })
+                    .then((data) => {
+                        const isVirtualContainer = Object.keys(data || {}).length === 0;
+
+                        if (isVirtualContainer && ref?.current) {
+                            ref.current.setAttribute("data-virtualcontainer", "true");
```

Comment:
If agreed we could prefix/scope the data attribute with cq for an outcome similar to `data-cq-virtualcontainer`

Reply:
Also, a unit test to set a contract would be welcomed

Reply:
Unit tests will be added when rewriting the repo, as we need to fix some failing tests adapt project first.